compiler/prelude/prelude.go

```diff
@@ -522,4 +522,48 @@ var $interfaceIsEqual = function(a, b) {
   }
   return $equal(a.$val, b.$val, a.constructor);
 };
+
+var $unsafeMethodToFunction = function(typ, name, isPtr) {
+  if (isPtr) {
+    return function(v) {
+      var r = v;
+      var ptrType = $ptrType(typ);
+      if (v.constructor != ptrType) {
+         switch (typ.kind) {
+         case $kindStruct:
+           r = $pointerOfStructConversion(v, ptrType);
+           break;
+         case $kindArray:
+           r = new ptrType(v);
+           break;
+         default:
+           r = new ptrType(v.$get,v.$set,v.$target);
+         }
+      }
+      return r[name](...[...arguments].slice(1));
+    }
```

Comment:
Since we can use ES 2015 syntax now, using rest parameters is [faster](https://jsbench.me/h5l8nnamor/1). Also I think we don't need separate variables for `r` and `v`.

```suggestion
    return function(r, ...args) {
      var ptrType = $ptrType(typ);
      if (r.constructor != ptrType) {
         switch (typ.kind) {
         case $kindStruct:
           r = $pointerOfStructConversion(r, ptrType);
           break;
         case $kindArray:
           r = new ptrType(r);
           break;
         default:
           r = new ptrType(r.$get, r.$set, r.$target);
         }
      }
      return r[name](...args);
    }
```

Same idea applied to the non-pointer branch.